build/commands/lib/util.js

```diff
@@ -606,7 +606,17 @@ const util = {
       util.run('goma_ctl', ['stat'], options)
     }
 
-    util.run('autoninja', ninjaOpts, options)
+    if (config.use_goma) {
+      // Setting `AUTONINJA_BUILD_ID` allows tracing Goma remote execution which helps with
+      // debugging issues (e.g., slowness or remote-failures).
+      console.log('Running build with ID ' + config.buildId)
+      options.env.AUTONINJA_BUILD_ID = config.buildId
+    }
+
+    // TODO(yannic): Switch this back to `autoninja` once upstream no longer unconditionally
+    // overrides `AUTONINJA_BUILD_ID`:
+    // https://source.chromium.org/chromium/chromium/tools/depot_tools/+/main:autoninja;l=14;drc=5506fbfba0c9cdf5d525f86fd2aa829e32c4c960
```

Comment:
why do this if `autoninja` does exactly this? If the only reason is to print this to console, then how about a `depot_tools` upstream to enable this conditionally with some env var?
We currently use `autoninja` because it sets `-j` more appropriately on high core (64+) machines (without goma).

Reply:
Printing the build ID will allow us to dig into issues with that build specifically. I think it's unlikely that upstream will accept a change to print the build ID from auto ninja.

I landed the change to not override the build ID if already defined upstream, so we can change this back to `autoninja`. How's `depot_tools` typically updated for Brave?

Reply:
> I landed the change to not override the build ID if already defined upstream, so we can change this back to `autoninja`

amazing! As I understand, `depot_tools` is updates itself automatically on any `gclient` call, so we should see it updated after `npm run sync`.

Reply:
Done, thanks!